File: app/src/main/java/com/rokudo/xpense/fragments/login/UserDetailsFragment.java
package com.rokudo.xpense.fragments.login;

import static com.rokudo.xpense.utils.DatabaseUtils.usersRef;

import android.content.Intent;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.google.firebase.auth.FirebaseAuth;
import com.rokudo.xpense.R;
import com.rokudo.xpense.activities.MainActivity;
import com.rokudo.xpense.databinding.FragmentUserDetailsBinding;


public class UserDetailsFragment extends Fragment {
    private static final String TAG = "UserDetailsFragment";

    private FragmentUserDetailsBinding binding;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        binding = FragmentUserDetailsBinding.inflate(inflater, container, false);


        binding.doneBtn.setOnClickListener(view1 -> {
            if (binding.displayNameTextInput.getText() == null || binding.displayNameTextInput.getText().toString().trim().equals("")) {
                Toast.makeText(requireContext(), "Name cannot be empty", Toast.LENGTH_SHORT).show();
            } else {
                if (FirebaseAuth.getInstance().getCurrentUser() != null && FirebaseAuth.getInstance().getCurrentUser().getPhoneNumber() != null) {
                    usersRef.document(FirebaseAuth.getInstance().getCurrentUser().getPhoneNumber())
                            .update("name", binding.displayNameTextInput.getText().toString())
                            .addOnSuccessListener(unused -> {
                                Intent i = new Intent(requireActivity(), MainActivity.class);
                                startActivity(i);
                                requireActivity().finish();
                            });
                }
            }

        });

        return binding.getRoot();
    }
}

File: app/src/main/java/com/rokudo/xpense/fragments/login/UserDetailsFragment.kt
package com.rokudo.xpense.fragments.login

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.compose.runtime.*
import androidx.compose.ui.platform.ComposeView
import androidx.fragment.app.Fragment
import com.google.firebase.auth.FirebaseAuth
import com.rokudo.xpense.activities.MainActivity
import com.rokudo.xpense.utils.DatabaseUtils
import com.rokudo.xpense.ui.theme.XpenseTheme

class UserDetailsFragment : Fragment() {

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        return ComposeView(requireContext()).apply {
            setContent {
                XpenseTheme {
                var name by remember { mutableStateOf("") }
                var isLoading by remember { mutableStateOf(false) }

                UserDetailsScreen(
                    name = name,
                    isLoading = isLoading,
                    onNameChange = { name = it },
                    onDone = {
                        if (name.trim().isNotEmpty()) {
                            isLoading = true
                            val currentUser = FirebaseAuth.getInstance().currentUser
                            val phoneNumber = currentUser?.phoneNumber

                            if (phoneNumber != null) {
                                DatabaseUtils.usersRef.document(phoneNumber)
                                    .update("name", name.trim())
                                    .addOnSuccessListener {
                                        isLoading = false
                                        startActivity(Intent(requireActivity(), MainActivity::class.java))
                                        requireActivity().finish()
                                    }
                                    .addOnFailureListener {
                                        isLoading = false
                                        Toast.makeText(
                                            requireContext(),
                                            "Failed to save name",
                                            Toast.LENGTH_SHORT
                                        ).show()
                                    }
                            }
                        } else {
                            Toast.makeText(
                                requireContext(),
                                "Name cannot be empty",
                                Toast.LENGTH_SHORT
                            ).show()
                        }
                    }
                )
                } // XpenseTheme
            }
        }
    }
}

